R/utils-xportr.R

```diff
@@ -242,14 +242,42 @@ xpt_validate <- function(data) {
     paste("datetime", 7:40, ".", sep = ""),
     paste("yymmdd", 2:10, ".", sep = ""),
     paste("mmddyy", 2:10, ".", sep = ""),
-    paste("ddmmyy", 2:10, ".", sep = "")
+    paste("ddmmyy", 2:10, ".", sep = ""),
+    "E8601DAw.",
+    "E8601DNw.",
+    "E8601TMw.d",
+    "E8601TZw.d",
+    "E8601LZw.d",
+    "E8601TXw.",
+    "E8601DTw.d",
+    "E8601DZw.d",
+    "E8601LXw.",
+    "E8601DXw.",
+    "IS8601DT.",
+    "E8601DT.",
+    "B8601DT.",
+    "IS8601DA.",
+    "B8601DA.",
+    "weekdate.",
+    "mmddyy.",
+    "ddmmyy.",
+    "yymmdd.",
+    "date.",
+    "time.",
+    "hhmm.",
+    "IS8601TM.",
+    "E8601TM.",
+    "B8601TM."
   )
```

Comment:
I ran the following code with the SAS format DATE9. also used in pilot 3 metadata and still got a message when I ran xportr_write

```
adsl <- data.frame(
  USUBJID = c(1001, 1002, 1003),
  BRTHDT = c(1, 1, 2)
)

metacore <- data.frame(
  dataset = c("adsl", "adsl"),
  variable = c("USUBJID", "BRTHDT"),
  format = c(NA, "DATE9.")
)

adsl <- xportr_format(adsl, metacore)

xportr_write(adsl, "adsl.xpt")
```

I got this message:
![image](https://github.com/atorus-research/xportr/assets/69685640/1bdef079-2b8a-476a-99b0-77aad984d313)

I am sorry if I am not clear. For example, E8601LZw.d is actually not a SAS format w and d should be replace by a range of numbers. Those numbers can be find in SAS documentation.

I had a look to SAS documentation for few formats and all the possible values for these format are: 

E8601DAw.: E8601DA. E8601DA10.

E8601DNw. : E8601DN. E8601DN10.

E8601TMw.d: E8601TM.
                      E8601TM8.0 E8601TM8.1 E8601TM8.2 E8601TM8.3 E8601TM8.4 E8601TM8.5 E8601TM8.6
                      E8601TM9.0 E8601TM9.1 E8601TM9.2 E8601TM9.3 E8601TM9.4 E8601TM9.5 E8601TM9.6
                      ...
                     E8601TM15.0 E8601TM15.1 E8601TM15.2 E8601TM15.3 E8601TM15.4 E8601TM15.5 E8601TM15.6

E8601TZw.d: E8601TZ.
			E8601TZ9.0 E8601TZ9.1 E8601TZ9.2 E8601TZ9.3 E8601TZ9.4 E8601TZ9.5 E8601TZ9.6
			...
			E8601TZ20.0 E8601TZ20.1 E8601TZ20.2 E8601TZ20.3 E8601TZ20.4 E8601TZ20.5 E8601TZ20.6

E8601LZw.: E8601LZ. E8601LZ9. E8601LZ10. E8601LZ11.  ...  E8601LZ20.

E8601TXw.: E8601TX. E8601TX9. E8601TX10. E8601TX.11  ...  E8601TX20.

...      

To avoid adding all the formats with the numbers, we could just use the prefix or use regular expression.

What do you think?